source/length.js

```diff
@@ -3,21 +3,33 @@ import _isNumber from './internal/_isNumber';
 
 
 /**
- * Returns the number of elements in the array by returning `list.length`.
+ * Returns the number of elements in the array, set or map
+ * by returning `collection.length` or `collection.size`.
  *
  * @func
  * @memberOf R
  * @since v0.3.0
  * @category List
  * @sig [a] -> Number
- * @param {Array} list The array to inspect.
- * @return {Number} The length of the array.
+ * @param {Array|Set|Map} collection The collection to inspect.
```

Comment:
Our docs are often aspirational rather than pedantically correct, but should we specifically mention `Array|Set|Map` here?  The implementation has little to do with these types but only with having either a `length` or a `size` property.

Of course I'm not sure of the simplest way to document that.  But is there a reason to specifically list these types?

Reply:
Well, those are all the builtin type that the function in its current form intentionally supports.

> I don't think we should call out `function` specifically. (https://github.com/ramda/ramda/pull/2450#issuecomment-361025289).

If `Object` support was added, the param type could be `Collection`.

That's the best I came up with. Do you have a better param type in mind?

Reply:
I think the discussion that we never followed up on there might be better.  This should simply be removed... although there definitely is something to be said for making it work with `Map` **and** `Set` as well as current `Array` and `Function` (and `Int16Array`, etc., and `AsyncFunction`, and who knows how many others?)

The point, though, is that with the plethora of types that this matches, even built-in ones, should we call out those three?

Reply:
> I think the discussion that we never followed up on there might be better. This should simply be removed...

I'm not sure what you mean, please be more clear.

> `Int16Array`

Right, didn't know typed arrays have a length too, never used them.

Reply:
I was simply saying that the discussion there ended up with something like a consensus on removing `length` altogether.